src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -29,27 +35,32 @@
 
 (schema/defn clojurize :- Event
   [event :- WatchEvent
-   watch-path :- Path]
-  {:type (get event-type-mappings (.kind event))
-   :path (when-not (= StandardWatchEventKinds/OVERFLOW (.kind event))
-           (-> watch-path
-               (.resolve (.context event))
-               fs/file))})
-
-;; This is quite similar to the function above but it is more a direct
-;; conversion of the exact data available on a specific WatchEvent instance,
-;; only used for debugging.
-(defn clojurize-for-logging
-  [e]
-  {:context (.context e)
-   :count (.count e)
-   :kind (.kind e)})
-
-(defn pprint-events
-  [events]
-  (->> events
-       (map #(update % :path str))
-       ks/pprint-to-string))
+   watched-path :- Path]
+  (let [kind (get event-type-mappings (.kind event))
+        changed-path (when-not (= :unknown kind)
+                       (.context event))
+        full-path (when-not (= :unknown kind)
+                    (-> watched-path (.resolve changed-path) fs/file))]
+    {:type kind
+     :count (.count event)
+     :watched-path (fs/file watched-path)
+     :changed-path (fs/file changed-path)
```

Comment:
This is returning the wrong pathed `File`. This is an err on my part. I wasn't aware that [`me.raynes.fs/file`](http://raynes.github.io/fs/me.raynes.fs.html#var-file) always roots relative files in the CWD which is the wrong behavior (we want it to continue to be a relative path).